client/asset/btc/btc.go

```diff
@@ -5418,83 +5473,348 @@ func (btc *baseWallet) MaxFundingFees(numTrades uint32, feeRate uint64, options
 	return feeRate * txSize
 }
 
-// checkPendingTxs checks to see if the wallet has received any incoming
-// transactions that need to be added to the transaction history. It also
-// checks all the pending transactions to see if they have been mined into
-// a block, and if so, updates the transaction history to reflect the
-// block height.
-func (btc *intermediaryWallet) checkPendingTxs(tip uint64) {
-	txHistoryDB := btc.txDB()
-	if txHistoryDB == nil {
-		return
+func rpcTxFee(tx *ListTransactionsResult) uint64 {
+	if tx.Fee != nil {
+		// Fee always seems to be negative in btcwallet, but just
+		// in case.
+		if *tx.Fee < 0 {
+			return toSatoshi(-*tx.Fee)
+		}
+		return toSatoshi(*tx.Fee)
 	}
+	return 0
+}
 
-	synced, _, err := btc.SyncStatus()
+// idUnknownTx identifies the type and details of a transaction either made
+// or recieved by the wallet.
+func (btc *intermediaryWallet) idUnknownTx(tx *ListTransactionsResult) (asset.TransactionType, *string, *asset.BondTxInfo, uint64, uint64, error) {
```

Comment:
Did you consider returning  a `wallet.Transaction` or an `ExtendedWalletTx`? This is fine too, but please name the return values.